lib/secret-tasks.js

```diff
@@ -0,0 +1,253 @@
+'use strict';
+
+/**
+ * @module SecretTasks
+ */
+
+var Q = require('q');
+var _ = require('lodash');
+var logger = require('winston');
+var Credstash = require('nodecredstash');
+var yaddaSecret = require('@asymmetrik/yadda-secret');
+
+/**
+ * Generate Credstash credential store
+ * @param {object} options - user provided options for the credential store
+ * @param {object} options.DeploymentCenter - user provided options for the deployment center
+ * @param {string} options.DeploymentCenter.name - Deployment center name
+ * @param {string} options.DeploymentCenter.region - Deployment center region
+ * @param {object} options.DeploymentCenter.secret - Secret options
+ * @param {string} options.DeploymentCenter.secret.kmsKeyAlias - KMS Master Key Alias to encrypt/decrypt data
+ * @return {module.exports|Credstash}
+ */
+function getCredentialStore(options){
+	var tableDetails = getCredentialTableDetails(options);
+	var store = new Credstash({
+		table: tableDetails.name,
+		awsOpts: { region: tableDetails.region },
+		kmsKey: options.DeploymentCenter.secret.kmsKeyAlias,
+	});
+	store.__name = tableDetails.name;
+	return store;
+}
+
+/**
+ * Generate secret table name from deployment center
+ * @param {string} options.DeploymentCenter.name - Deployment center name
+ * @param {string} options.DeploymentCenter.region - Deployment center region
+ * @return {{name: string, region: string}}
+ */
+function getCredentialTableDetails(options){
+	return {
+		name: options.DeploymentCenter.name+'-Secrets',
+		region: options.DeploymentCenter.region,
+	}
+}
+
+/**
+ * Creates SecretCenter table in DynamoDB
+ * @param {object} options - user provided options
+ */
+exports.createSecretCenter = function(options){
+	return Q.when(options)
+		.then(function(){
+			logger.info('Creating DeploymentCenter Secret Center');
+			var store = getCredentialStore(options);
```

Comment:
Why wouldn't you want to just use `options.DeploymentCenter.secret.secretStore` here, since `setupSecretCenter` has already been called?  Is there an advantage to creating the store again?

Reply:
When I wrote this part I hadn't done the setupSecretCenter part yet. Good catch